### PR TITLE
Updating the DevTools docs pages

### PR DESCRIPTION
**Pull request by sfshaza2**
Staged:
[updated memory page](https://sz-flutter-2.web.app/development/tools/devtools/memory)
[new debug console page](https://sz-flutter-2.web.app/development/tools/devtools/console)

P.S. I'm aware that the staged site displays some of the new images in a huge size. Those will be fixed.

Please review, @polina-c

### CODE REVIEW

File: src/development/tools/devtools/console.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+{{site.alert.note}}
+  This page is up to date for DevTools 2.23.0.
+{{site.alert.end}}
+
+The Debug console is available from the [Inspector][],
+[Debugger][], and [Memory][] views. To bring up
+the debug console, click the **More info** link,
+as shown:
+
+![Screenshot of how to bring up Debug console]({{site.url}}/assets/images/docs/tools/devtools/debug-console.png){:width="100%"}
```
Comment (polina-c) [inline src/development/tools/devtools/console.md:21]: Please, remove this text and picture. This was explanation of the context for the tech writers on how user will access the documentation:

"To bring up
the debug console, click the **More info** link,
as shown:"

File: src/development/tools/devtools/console.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+## Evaluate expressions
+
+In the console, you can evaluate expressions for a paused
+or running application, assuming that you are running
+your app in debug mode:
+
+![Screenshot showing evaluating an expression in the console]({{site.url}}/assets/images/docs/tools/devtools/console-evaluate-expressions.png)
+
+To assign an evaluated object to a variable,
+use $0, $1 … $5:
```
Comment (polina-c) [inline src/development/tools/devtools/console.md:49]: Let's make it easier for people who cannot recognize text on images:

use $0, $1 … $5 in the form `var x = $0`:

Comment (sfshaza2) [reply src/development/tools/devtools/console.md:49]: Are they limited to 5?

**Comment by sfshaza2** [comment]
@polina-c, your feedback is incorporated and staging is updated. Please review again.

**Comment by sfshaza2** [comment]
@parlough, can you rubber stamp?

File: src/development/tools/devtools/console.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+![Screenshot showing how to evaluate variables]({{site.url}}/assets/images/docs/tools/devtools/console-evaluate-variables.png){:width="100%"}
+
+## Browse heap snapshot
+
+To drop a variable to the console from a heap snapshot,
+do the following:
+
+1. Navigate to **Devtools > Memory > Diff Snapshots**.
+1. Click on the context menu [`⋮]` to view the number of
```
Comment (parlough) [inline src/development/tools/devtools/console.md:55]: I'm not sure what this is, but is the opening bracket supposed to be in code syntax too?

```suggestion
1. Click on the context menu `[⋮]` to view the number of
```

File: src/development/tools/devtools/console.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+---
+title: Using the Debug console
+description: Learn how to use the DevTools console to xxx.
```
Comment (parlough) [inline src/development/tools/devtools/console.md:3]: The "to xxx" might not be understood by everyone. I think just dropping it is fine:

```suggestion
description: Learn how to use the DevTools console.
```

File: src/development/tools/devtools/memory.md [hunk 5]
```diff
@@ -145,23 +149,12 @@ root -> d -> e -> x (shortest retaining path to `x`)
...
-
-![Screenshot of a profile page]({{site.url}}/assets/images/docs/tools/devtools/profile-tab.png)
-
 ### Memory leaks happen in Dart?
 
 Garbage collector cannot prevent all types of memory leaks, and developers
 still need to watch objects to have leak-free lifecycle. 
 
-#### Why garbage collector cannot prevent all leaks?
+#### Why can't garbage collector prevent all leaks?
```
Comment (parlough) [inline src/development/tools/devtools/memory.md:157]:
```suggestion
#### Why can't the garbage collector prevent all leaks?
```

File: src/development/tools/devtools/memory.md [hunk 6]
```diff
@@ -274,14 +267,14 @@ has the following features:
   and view both standard events (like garbage collection)
   and custom events (like image allocation).
 
-[**Profile** tab](#profile-tab)
+[**Profile Memory** tab](#profile-tab)
```
Comment (parlough) [inline src/development/tools/devtools/memory.md:270]: Have to update these to link to the new headings:

```suggestion
[**Profile Memory** tab](#profile-memory-tab)
```

File: src/development/tools/devtools/memory.md [hunk 7]
```diff
@@ -274,14 +267,14 @@ has the following features:
...
 : See current memory allocation listed by class and
-  memory type (Dart or external/native). 
+  memory type. 
 
-[**Diff** tab](#diff-tab)
+[**Diff Snapshots** tab](#diff-tab)
```
Comment (parlough) [inline src/development/tools/devtools/memory.md:274]:
```suggestion
[**Diff Snapshots** tab](#diff-snapshots-tab)
```

File: src/development/tools/devtools/memory.md [hunk 8]
```diff
@@ -274,14 +267,14 @@ has the following features:
...
 : Detect and investigate a feature’s memory management issues.
 
-[**Trace** tab](#trace-tab)
+[**Trace Instances** tab](#trace-tab)
```
Comment (parlough) [inline src/development/tools/devtools/memory.md:277]:
```suggestion
[**Trace Instances** tab](#trace-instances-tab)
```

File: src/development/tools/devtools/console.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+## Evaluate expressions
+
+In the console, you can evaluate expressions for a paused
+or running application, assuming that you are running
+your app in debug mode:
+
+![Screenshot showing evaluating an expression in the console]({{site.url}}/assets/images/docs/tools/devtools/console-evaluate-expressions.png)
+
+To assign an evaluated object to a variable,
+use $0, $1 (through $5) in the form of `var x = $0`:
```
Comment (parlough) [inline src/development/tools/devtools/console.md:45]: Optionally put these in code syntax:

```suggestion
use `$0`, `$1` (through `$5`) in the form of `var x = $0`:
```

**Comment by polina-c** [comment]
Something wrong here:

Screenshot 2023-04-13 at 1 34 56 PM

**Comment by polina-c** [comment]
> @polina-c, your feedback is incorporated and staging is updated. Please review again.

Just one more comment.

**Comment by sfshaza2** [comment]
All fixed, @polina-c and @parlough! Let's land this puppy!

**Review by parlough — APPROVED** [review]

**Pull request merged**